metric_learn/covariance.py

```diff
@@ -35,11 +35,11 @@ def fit(self, X, y=None):
     y : unused
     """
     X = self._prepare_inputs(X, ensure_min_samples=2)
-    M = np.cov(X, rowvar = False)
-    if M.ndim == 0:
+    M = np.atleast2d(np.cov(X, rowvar = False))
+    if len(M) == 1:
```

Comment:
Because of the atleast2d, this needs to change

Reply:
Can we be more explicit and check `M.size == 1`? Would it be possible for M to be a column vector here?

Reply:
Ah yes you're right thanks
I thought that len counted the number of elements but indeed `len(np.array([[1, 2, 3]])) == 1`

Reply:
done